extensions/dim-rename-kafka8/src/main/java/io/druid/server/namespace/KafkaExtractionManager.java

```diff
@@ -0,0 +1,231 @@
+
+  private final Properties kafkaProperties;
+  private final Collection<ListenableFuture<?>> futures = new ConcurrentLinkedQueue<>();
+  private final ListeningExecutorService executorService = MoreExecutors.listeningDecorator(
+      Executors.newCachedThreadPool(
+          new ThreadFactoryBuilder()
+              .setNameFormat("kafka-rename-consumer-%d")
+              .setDaemon(true)
+              .setPriority(Thread.MIN_PRIORITY)
+              .build()
+      )
+  );
+  private final AtomicInteger startedPipes = new AtomicInteger(0);
+  private final AtomicLong events = new AtomicLong(0l);
+
+  // Bindings in KafkaExtractionNamespaceModule
+  @Inject
+  public KafkaExtractionManager(
+      Injector injector
+  )
+  {
+    if (!NamespacedExtractionModule.isNamespaceEnabled(injector)) {
+      this.kafkaProperties = null;
+    } else {
+      final Properties kafkaProperties = injector.getInstance(
+          Key.get(
+              Properties.class,
+              Names.named("renameKafkaProperties")
+          )
+      );
+      this.kafkaProperties = new Properties();
+      if (kafkaProperties != null) {
+        if (kafkaProperties.containsKey("group.id")) {
+          throw new IAE(
+              "Cannot set kafka property [group.id]. Property is randomly generated for you. Found [%s]",
+              kafkaProperties.getProperty("group.id")
+          );
+        }
+        if (kafkaProperties.containsKey("auto.offset.reset")) {
+          throw new IAE(
+              "Cannot set kafka property [auto.offset.reset]. Property will be forced to [smallest]. Found [%s]",
+              kafkaProperties.getProperty("auto.offset.reset")
+          );
+        }
+        this.kafkaProperties.putAll(kafkaProperties);
+      }
+      if (!this.kafkaProperties.containsKey("zookeeper.connect")) {
+        this.kafkaProperties.put("zookeeper.connect", "localhost:2181/kafka");
+      }
+      // Enable publish-subscribe
+      this.kafkaProperties.setProperty("auto.offset.reset", "smallest");
+    }
+  }
+
+  public long getBackgroundTaskCount()
+  {
+    return startedPipes.get();
+  }
+
+  private static final Decoder<String> defaultStringDecoder = new Decoder<String>()
+  {
+    @Override
+    public String fromBytes(byte[] bytes)
+    {
+      return StringUtils.fromUtf8(bytes);
+    }
+  };
+
+  public long getNumEvents()
+  {
+    return events.get();
+  }
+
+  private static void updateMap(
+      final String topic,
+      final MessageAndMetadata<String, String> messageAndMetadata,
+      final Map<String, String> map
+  )
+  {
+    final String key = messageAndMetadata.key();
+    final String message = messageAndMetadata.message();
+    if (key == null || message == null) {
+      log.error("Bad key/message from topic [%s]: [%s]", topic, messageAndMetadata);
+      return;
+    }
+    map.put(key, message);
+  }
+
+  public void addListener(final KafkaExtractionNamespace kafkaNamespace, final Map<String, String> map)
```

Comment:
Here it only has a put. In the namespace workflow it has a get. And I don't remember if it uses a "contains". But since there are lots of things which conform to Map out of the box, I decided to use that. This is in contrast to the namespace extraction function finder which worked out better (IMHO) with a `Function<String, Function<String, String>>`